src/goto-synthesizer/goto_synthesizer_parse_options.cpp

```diff
@@ -61,8 +66,45 @@ int goto_synthesizer_parse_optionst::doit()
     configure_gcc(gcc_version);
   }
 
-  // Get options and preprocess `goto_model`.
-  const auto &options = get_options();
+  optionalt<cmdlinet> cbmc_cmdline;
+  if(cmdline.isset("cbmc-cmd"))
+  {
+    std::string cbmc_cmdline_str = cmdline.get_value("cbmc-cmd");
+
+    // Tokenize cbmc_cmdline_str
+    std::vector<char *> cbmc_args;
+    std::istringstream iss(cbmc_cmdline_str);
+    std::string token;
+    while(iss >> token)
+    {
+      char *cbmc_arg = new char[token.size() + 1];
+      copy(token.begin(), token.end(), cbmc_arg);
+      cbmc_arg[token.size()] = '\0';
+      cbmc_args.push_back(cbmc_arg);
+    }
+
+    // cbmc-cmd must starts with `cbmc`
+    if(strcmp(cbmc_args[0], "cbmc") != 0)
+    {
+      throw invalid_command_line_argument_exceptiont(
+        "cbmc-cmd must starts with cbmc, e.g., --cbmc-cmd \"cbmc [args]\"",
```

Comment:
I don't have much background, but why not make the option `--cbmc-options` instead to remove the need to specify `cbmc` (which seems redundant?).

Reply:
`goto-synthesizer` checks its arguments of  that start with `--` against the `SYNTHEZIER_OPT`, and errors out if the argument is not matched any defined option. It will happen if some CBMC option is not one of the synthesizer options.
So we let the argument start with `cbmc` instead to avoid the check.